zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java

```diff
@@ -142,6 +142,29 @@ public CompletionStage<Collection<Document>> getFlowNodeInstances(
     return fetchUnboundedDocumentCollection(request, hit -> new Document(hit.id(), hit.index()));
   }
 
+  @Override
+  public CompletionStage<Collection<ProcessInstanceDocument>> getProcessInstances(
+      final List<String> processInstanceIds) {
+    final var idQ = QueryBuilders.ids(i -> i.values(processInstanceIds));
+    final var typeQ =
+        QueryBuilders.term(
+            t ->
+                t.field(ListViewTemplate.JOIN_RELATION)
+                    .value(ListViewTemplate.PROCESS_INSTANCE_JOIN_RELATION));
+    final var request =
+        new SearchRequest.Builder()
+            .index(listViewAlias)
+            .source(s -> s.filter(f -> f.includes(ListViewTemplate.TREE_PATH)))
+            .query(q -> q.bool(b -> b.must(idQ, typeQ)));
+
+    return fetchUnboundedDocumentCollection(
+        request,
+        ProcessInstanceForListViewEntity.class,
+        hit ->
+            new ProcessInstanceDocument(
+                hit.id(), hit.index(), Long.parseLong(hit.id()), hit.source().getTreePath()));
```

Comment:
:thought_balloon:  do we need to handle potential errors when applying the transformer? 
 `Long.parseLong` may throw an exception in the transformer.

We have three options:
- throw an exception: the future may never be completed and the exception is bubbled up
- we fail the future
- we may also want to skip that document, log the exception and keep processing the rest of the documents 

I don't have enough context to know what's the better approach here. 
We don't necessarily need to discuss it in this PR.

Reply:
I believe we do handle the future when the exception is thrown, but assuming it's deterministic, we would get stuck there.

As this was the previous behavior, I would assume this is ok. That said, it's on my improvement list to not use the ID but instead the key from the document (which is already a long).